java/src/org/openqa/selenium/firefox/AddHasExtensions.java

```diff
@@ -65,9 +72,24 @@ public HasExtensions getImplementation(Capabilities capabilities, ExecuteMethod
       public String installExtension(Path path) {
         Require.nonNull("Extension Path", path);
 
-        return (String) executeMethod.execute(
-          INSTALL_EXTENSION,
-          ImmutableMap.of("path", path.toAbsolutePath().toString(), "temporary", false));
+        try {
+          return (String) executeMethod.execute(
+            INSTALL_EXTENSION,
+            ImmutableMap.of("path", path.toAbsolutePath().toString(), "temporary", false));
+        } catch (WebDriverException ex) {
+          File localFile = new LocalFileDetector().getLocalFile(path.toString());
```

Comment:
The more correct thing to do would be to make `RemoteExecuteMethod` implement `WrapsDriver`. You could then get the driver, check if it's remote, and then get the file detector that way. Relying on exceptions for flow control is generally frowned upon.